### PR TITLE
Staking Reward tests Round 2

### PR DESCRIPTION
**Pull request by skellet0r**
Here we go again :)))

Fixed minor bugs in the Reward Stream stateful test (I was setting a value in a for loop, and then updating that variable, and making an assertion based on the value from the for loop .... duh gotta read from the update)

Also added some tests for the new set distributor and transfer ownership fns.

### CODE REVIEW

File: tests/RewardStream/test_ownership_change.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import brownie
+import pytest
+
+
+@pytest.mark.parametrize("commit,accept", [(a, b) for a in [0, 1] for b in [0, 1]])
```
Comment (iamdefinitelyahuman) [inline tests/RewardStream/test_ownership_change.py]: this looks like a job for `itertools.combinations`

File: tests/integration/test_reward_stream_state.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+import brownie
+from brownie import RewardStream, chain
+from brownie.test import strategy
+from brownie_tokens import ERC20
+
+DAY = 60 * 60 * 24
+
+
+class _State:
```
Comment (iamdefinitelyahuman) [inline tests/integration/test_reward_stream_state.py:11]: Woah, impressive!

**Review by iamdefinitelyahuman — APPROVED** [review]

File: tests/RewardStream/test_ownership_change.py (hunk 1 above)
Comment (skellet0r) [reply tests/RewardStream/test_ownership_change.py]: yeah you're right, I was lazy and didn't want to import, but makes to code cleaner to just use itertools.product

**Comment by skellet0r** [comment]
Gotta merge #9 first and then I'll force push to get the CI to run these tests

**Comment by iamdefinitelyahuman** [comment]
> Gotta merge #9 first and then I'll force push to get the CI to run these tests

Merged, bls rebase

**Review by iamdefinitelyahuman — APPROVED** [review]

**Pull request merged**